apps/api/src/server/utils/__tests__/published-case-fields-changed.test.js

```diff
@@ -0,0 +1,166 @@
+				easting: 111111,
+				northing: 222222
+			},
+			applicant: {}
+		};
+		updated = cloneDeep(original);
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	describe('Should return false', () => {
+		test('when nothing has changed', () => {
+			expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(false);
+		});
+
+		test('when only the modifiedAt field has changed', () => {
+			updated.modifiedAt = Date.now();
+			expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(false);
+		});
+
+		test('when only the modifiedAt, createdAt and id fields in the CaseStatus have changed', () => {
+			updated.CaseStatus.modifiedAt = Date.now();
+			updated.CaseStatus.createdAt = Date.now();
+			updated.CaseStatus.id++;
+			expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(false);
+		});
+	});
+
+	describe('Should return true', () => {
+		describe('when checking project information', () => {
+			test('and the Project name has changed', () => {
+				updated.title = 'Project name';
+				expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(true);
+			});
+
+			test('and the Case Stage has changed', () => {
+				updated.CaseStatus[0].status = 'Case Stage';
+				expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(true);
+			});
+
+			test('and the Project description has changed', () => {
+				updated.description = 'Project description';
+				expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(true);
+			});
+
+			test('and the Project email address has changed', () => {
+				updated.ApplicationDetails.caseEmail = 'email@pins-test.gov';
+				expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(true);
+			});
+
+			test('and the Project location has changed', () => {
+				updated.ApplicationDetails.locationDescription = 'Project Location';
+				expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(true);
+			});
+
+			test('and the Grid references for northing has changed', () => {
+				updated.gridReference.northing = 123456;
+				expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(true);
+			});
+
+			test('and the Grid references for easting has changed', () => {
+				updated.gridReference.easting = 123456;
+				expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(true);
+			});
+
+			test('and the Regions have changed', () => {
+				updated.ApplicationDetails.regions.push({ regionId: 30 });
+				expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(true);
+			});
+
+			test('and the Map zoom level has changed', () => {
+				updated.ApplicationDetails.zoomLevel.name = 'X';
+				expect(publishedCaseFieldsHaveChanged(original, updated)).toBe(true);
+			});
+		});
+
+		describe('when checking applicant information', () => {
+			[
+				{ field: 'organisationName', description: 'Organisation name' },
+				{ field: 'website', description: 'Website' },
+				{ field: 'email', description: 'Email address' }
+			].map(({ field, description }) =>
```

Comment:
**nit-pick:** `.forEach` makes more sense here as the return value of `.map` isn't used.

```suggestion
			].forEach(({ field, description }) =>
```

Reply:
Changed map to forEach as suggested